cpp/src/arrow/gpu/cuda_context.cc

```diff
@@ -274,7 +343,18 @@ Status CudaContext::OpenIpcBuffer(const CudaIpcMemHandle& ipc_handle,
   return Status::OK();
 }
 
+Status CudaContext::CloseIpcBuffer(CudaBuffer* buf) {
+  ContextSaver set_temporary((CUcontext)(handle()));
```

Comment:
Ditto.

Reply:
Fixed.
Btw, I had to change
  `const void* CudaContext::handle() const`
to
  `void* CudaContext::handle() const`
to make 
  `reinterpret_cast<CUcontext>(handle())`
to compile.

Reply:
👍